packages/dtslint/src/index.ts

```diff
@@ -1,22 +1,31 @@
 #!/usr/bin/env node
 
+import { getTypesVersions } from "@definitelytyped/header-parser";
 import { AllTypeScriptVersion, TypeScriptVersion } from "@definitelytyped/typescript-versions";
-import assert = require("assert");
+import { assertNever, deepEquals, readJson } from "@definitelytyped/utils";
 import fs from "fs";
+import os from "os";
 import { basename, dirname, join as joinPaths, resolve } from "path";
+import { satisfies } from "semver";
+import { pipeline } from "stream/promises";
+import tar from "tar";
+import { createGunzip } from "zlib";
+import { checkPackageJson, checkTsconfig, runAreTheTypesWrong } from "./checks";
+import { TsVersion, lint } from "./lint";
+import { getCompilerOptions, packageDirectoryNameWithVersionFromPath, packageNameFromPath } from "./util";
+import assert = require("assert");
 
-import { deepEquals } from "@definitelytyped/utils";
-import { checkPackageJson, checkTsconfig } from "./checks";
-import { lint, TsVersion } from "./lint";
-import { getCompilerOptions, packageNameFromPath } from "./util";
-import { getTypesVersions } from "@definitelytyped/header-parser";
+const tmpDir = os.tmpdir();
```

Comment:
I would suggest using the `tmp` library for these operations and creating one tmpdir with it; the library can ensure that the folder is cleared on exit and auto manages creating unique names so parallel processes don't have problems.